controllers/ocimachine_controller.go

```diff
@@ -115,16 +116,40 @@ func (r *OCIMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		r.Recorder.Eventf(ociMachine, corev1.EventTypeWarning, "ClusterNotAvailable", "Cluster is not available yet")
 		return ctrl.Result{}, nil
 	}
-
 	regionOverride := r.Region
+
+	var clients scope.OCIClients
+	var clientProvider *scope.ClientProvider
+	if ociCluster.Spec.IdentityRef != nil {
+		identity, err := cloudutil.GetClusterIdentityFromRef(ctx, r.Client, ociCluster.Namespace, ociCluster.Spec.IdentityRef)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if !cloudutil.IsClusterNamespaceAllowed(ctx, r.Client, identity.Spec.AllowedNamespaces, ociCluster.Namespace) {
+			conditions.MarkFalse(ociCluster, infrastructurev1beta1.ClusterReadyCondition, infrastructurev1beta1.NamespaceNotAllowedByIdentity, clusterv1.ConditionSeverityError, "")
+			return reconcile.Result{}, errors.New("OCIClusterIdentity list of allowed namespaces doesn't include current cluster namespace")
+		}
+		clientProvider, err = cloudutil.GetOrBuildClientFromIdentity(ctx, r.Client, identity, regionOverride)
+		if err != nil {
+			logger.Error(err, "Could not create client provider for the cluster identity")
+			return reconcile.Result{}, err
+		}
+		region, err := clientProvider.GetRegion()
+		if err != nil {
+			logger.Error(err, "Could not get region from cluster identity")
+			return reconcile.Result{}, err
+		}
+		regionOverride = region
+	} else {
+		clientProvider = r.ClientProvider
+	}
```

Comment:
This code could be moved into a reusable method since it is also in `controllers/ocicluster_controller.go`